app/src/main/java/com/neudesic/myapplication/domain/useCase/GetDadJokesUseCase.kt

```diff
@@ -0,0 +1,8 @@
+package com.neudesic.myapplication.domain.useCase
+
+import com.neudesic.myapplication.domain.repository.DadJokeRepository
+import javax.inject.Inject
+
+class GetDadJokesUseCase @Inject constructor(private val dadJokeRepository: DadJokeRepository) {
```

Comment:
I would suggest abstracting the "UseCase" functionality where you have an invoke or execute function. Then rename the class `GetSingleDadJokeUseCase`. So your call would look like `getSingleDadJokeUseCase.invoke()`. Reference: https://proandroiddev.com/why-you-need-use-cases-interactors-142e8a6fe576

Reply:
what if i want to support multiple cases.
like `getSingleDadJoke` and `getDadJokes`

might be a bit extra to break them down further into each own file, rather than `GetDadJokesUseCase`

I saw the article you linked, but his use case is all 1 thing.
here's another one talking about grouping it

https://proandroiddev.com/the-neverending-use-case-story-d30440e5c7f0

Reply:
Those are separate use cases, no? You mean have several use cases (classes) in one file? In an app like this, I would say that sounds good to me.